src/Microsoft.Health.Dicom.Core/Extensions/DicomElementExtensions.cs

```
@@ -0,0 +1,146 @@
+            { DicomVR.AS, GetStringValue },
+            { DicomVR.AT, GetUIntValue },
+            { DicomVR.CS, GetStringValue },
+            { DicomVR.DA, GetDAValue },
+            { DicomVR.DS, GetStringValue },
+            { DicomVR.DT, GetDTValue },
+            { DicomVR.FL, GetFloatValue },
+            { DicomVR.FD, GetDoubleValue },
+            { DicomVR.IS, GetStringValue },
+            { DicomVR.LO, GetStringValue },
+            { DicomVR.PN, GetStringValue },
+            { DicomVR.SH, GetStringValue },
+            { DicomVR.SL, GetIntValue },
+            { DicomVR.SS, GetShortValue },
+            { DicomVR.TM, GetTMValue },
+            { DicomVR.UI, GetStringValue },
+            { DicomVR.UL, GetUIntValue },
+            { DicomVR.US, GetUShortValue },
+        };
+
+        public static object GetSingleValue(this DicomElement element)
+        {
+            EnsureArg.IsNotNull(element, nameof(element));
+            if (element.Count != 1)
+            {
+                return null;
+            }
+
+            if (!Readers.ContainsKey(element.ValueRepresentation))
+            {
+                Debug.Fail($"Need reader for VR {element.ValueRepresentation}");
+                return null;
+            }
+
+            return Readers[element.ValueRepresentation].Invoke(element);
+        }
+
+        private static object GetDAValue(DicomElement element)
+        {
+            string stringDate = (string)SafeGetValue<string>(element);
+
+            if (stringDate == null ||
+                !DicomElementMinimumValidation.TryParseDA(stringDate, out DateTime result))
+            {
+                return null;
+            }
+
+            return result;
+        }
+
+        private static object SafeGetValue<T>(DicomElement element)
+        {
+            try
+            {
+                return element.Get<T>();
+            }
+            catch
+            {
+                return null;
```

add log

---

how to add log for static methods?

---

make it instance class?

---

You could add the option of passing a logger

---

Or you could have a statically-scoped `ILoggerFactory<T>` that was configured in start-up that this class can use. Given this is a library, it is a bit more difficult to do that though.

---

However, should we really be swallowing type-mismatch exceptions like this?

---

This class is removed